src/OhadSoft.AzureLetsEncrypt.Renewal/Management/RenewalParameters.cs

```diff
@@ -104,7 +104,13 @@ public static int VerifyRenewXNumberOfDaysBeforeExpiration(int renewXNumberOfDay
 
         public override string ToString()
         {
-            return Invariant($"{nameof(WebApp)}: {WebApp}, {nameof(Hosts)}: {String.Join(", ", Hosts)}, {nameof(ServicePlanResourceGroup)}: {ServicePlanResourceGroup}, {nameof(SiteSlotName)}: {SiteSlotName}, {nameof(GroupName)}: {GroupName}, {nameof(WebAppEnvironmentParams)}: {WebAppEnvironmentParams}, {nameof(AzureDnsEnvironmentParams)}: {AzureDnsEnvironmentParams}, {nameof(AzureDnsZoneName)}: {AzureDnsZoneName}, {nameof(AzureDnsRelativeRecordSetName)}: {AzureDnsRelativeRecordSetName}, {nameof(UseIpBasedSsl)}: {UseIpBasedSsl}, {nameof(RsaKeyLength)}: {RsaKeyLength}, {nameof(AcmeBaseUri)}: {AcmeBaseUri}, {nameof(WebRootPath)}: {WebRootPath}, {nameof(RenewXNumberOfDaysBeforeExpiration)}: {RenewXNumberOfDaysBeforeExpiration}, {nameof(AuthenticationUri)}: {AuthenticationUri}, {nameof(AzureTokenAudience)}: {AzureTokenAudience}, {nameof(AzureManagementEndpoint)}: {AzureManagementEndpoint}, {nameof(AzureDefaultWebsiteDomainName)}: {AzureDefaultWebsiteDomainName}");
+            return JsonConvert.SerializeObject(
```

Comment:
This will expose the service principal client secret in the logs!

Reply:
the [JsonIgnore] hint above the ClientSecret property prevents it from being serialized and printed